TP3/src/main/java/ar/edu/itba/sia/GeneticOperators/Selections/UniversalSelection.java

```diff
@@ -1,23 +1,39 @@
 package ar.edu.itba.sia.GeneticOperators.Selections;
 
-import ar.edu.itba.sia.GeneticOperators.Interfaces.Selection;
+import ar.edu.itba.sia.GeneticOperators.Interfaces.CustomizableSelection;
 import ar.edu.itba.sia.Warriors.Warrior;
+import ar.edu.itba.sia.util.FitnessUtils;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class UniversalSelection implements Selection {
-    public List<Warrior> select(List<Warrior> warriors, int quantity) {
+public class UniversalSelection implements CustomizableSelection {
+
+    /**
+     * Select warriors as per universal selection, optionally allowing for custom fitnesses instead of {@link Warrior#getFitness()}
+     * (eg. for Ranking or Boltzmann).
+     *
+     * @param warriors        Warriors to select from
+     * @param quantity        Number of warriors to select
+     * @param customFitnesses (Optional) Custom fitnesses to use. If null, will use warriors' fitness. If provided, should
+     *                        correspond to each warrior's custom fitness by index.
+     * @return The selected warriors.
+     */
+    public List<Warrior> select(List<Warrior> warriors, int quantity, List<Double> customFitnesses) {
+        if (customFitnesses != null && customFitnesses.size() != warriors.size()) {
+            throw new IllegalArgumentException(String.format("Warriors and custom fitnesses must match in length. %d != %d", warriors.size(), customFitnesses.size()));
+        }
+
         double r = Math.random();
         double[] randoms = new double[quantity];
         initializeRandoms(randoms, quantity, r);
         List<Warrior> selectedWarriors = new ArrayList<>();
         int warriorIndex = 0;
         int randomIndex  = 0;
-        double totalFitness = getTotalFitness(warriors);
+        double totalFitness = FitnessUtils.getTotalFitness(warriors);
```

Comment:
Acá hay que sumar las custom fitnesses si nos las dan, no? En vez de los fitnesses de los warriors

```suggestion
        double totalFitness = customFitnesses == null ? FitnessUtils.getTotalFitness(warriors) : FitnessUtils.getTotalCustomFitness(customFitnesses)
```